course7/homeworks/waleed/complex/main.cpp

```diff
@@ -0,0 +1,27 @@
+#include "complex.h"
+#include "calculator.h"
+
+#include <string>
+#include <iostream>
+#include <fstream>
+
+int main()
+{
+    std::string inp;
+    Calculator calc;
+    while (std::getline(std::cin, inp))
+    {
+        calc.PushLine(inp);
+    }
+
+    try
+    {
+        std::cout << calc.GetTop() << std::endl;
+    }
+    catch (std::exception)
```

Comment:
It is not a good idea to use exceptions for control flow, as they are horribly slow to throw in C++.

Maybe you could design the API of ```GetTop``` so that it returns a pointer to Complex, or null if stack is empty?

Reply:
I agree on the performance issue for exceptions, but the problem with using pointer to fix it is that if I return address of an element in the Calculator's stack, it might be changed in the future (by pops and pushes) without the holder of that address noticing. 
And if I create a new instance in the heap and return its address, who should be responsible for destroying it?